sample/src/main/kotlin/com/slack/circuit/sample/home/CompositeScreen.kt

```diff
@@ -0,0 +1,194 @@
+import com.slack.circuit.sample.petlist.PetList
+import com.slack.circuit.sample.petlist.PetListPresenter
+import com.slack.circuit.sample.petlist.PetListScreen
+import com.slack.circuit.ui
+import com.squareup.anvil.annotations.ContributesMultibinding
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import javax.inject.Inject
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.map
+import kotlinx.parcelize.Parcelize
+
+// Need better naming conventions.
+@Parcelize
+object CompositeScreen : Screen {
+  data class CompositeState(val homeState: HomeScreen.State, val petListState: PetListScreen.State)
+
+  sealed interface CompositeEvent {
+    class CompositeHomeEvent(val event: HomeScreen.Event.NavClickEvent) : CompositeEvent
+    class CompositePetListEvent(val event: PetListScreen.Event) : CompositeEvent
+  }
+}
+
+@ContributesMultibinding(AppScope::class)
+class CompositePresenterFactory
+@Inject
+constructor(private val compositePresenterFactory: CompositePresenter.Factory) : PresenterFactory {
+  override fun create(screen: Screen, navigator: Navigator): Presenter<*, *>? {
+    if (screen is CompositeScreen) return compositePresenterFactory.create(navigator)
+    return null
+  }
+}
+
+class CompositePresenter
+@AssistedInject
+constructor(
+  @Assisted private val navigator: Navigator,
+  private val homePresenter: HomePresenter,
+  petListPresenterFactory: PetListPresenter.Factory
+) : Presenter<CompositeScreen.CompositeState, CompositeScreen.CompositeEvent> {
+  private val petListPresenter = petListPresenterFactory.create(navigator)
+
+  @SuppressLint("FlowOperatorInvokedInComposition")
+  @Composable
+  override fun present(
+    events: Flow<CompositeScreen.CompositeEvent>
+  ): CompositeScreen.CompositeState {
+    val homeState =
+      homePresenter.present(
+        events.filterIsInstance<CompositeScreen.CompositeEvent.CompositeHomeEvent>().map {
+          it.event
+        }
+      )
+    val petListState =
+      petListPresenter.present(
+        events.filterIsInstance<CompositeScreen.CompositeEvent.CompositePetListEvent>().map {
+          it.event
+        }
+      )
+
+    return CompositeScreen.CompositeState(homeState, petListState)
+  }
+
+  @AssistedFactory
+  interface Factory {
+    fun create(navigator: Navigator): CompositePresenter
+  }
+}
+
+@ContributesMultibinding(AppScope::class)
+class CompositeScreenFactory @Inject constructor() : ScreenViewFactory {
+  override fun createView(screen: Screen): ScreenView? {
+    if (screen is CompositeScreen) {
+      return ScreenView(compositeUi())
+    }
+    return null
+  }
+}
+
+private fun compositeUi() =
+  ui<CompositeScreen.CompositeState, CompositeScreen.CompositeEvent> { state, events ->
+    CompositeScreen(state, events)
+  }
+
+@Composable
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+fun CompositeScreen(
```

Comment:
I think having this name collide with the screen key is bad and we should do something diff. Maybe `Home` or `HomeContent`?